src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,482 @@
+    ///     "Paris is a city in France."
+    /// ];
+    /// let output = ner_model.predict(&input, true);
+    ///# Ok(())
+    ///# }
+    /// ```
+    pub fn predict(&self, input: &[&str], ignore_first_label: bool) -> Vec<Token> {
+        let input_tensor = self.prepare_for_model(input.to_vec());
+        let (output, _, _) = no_grad(|| {
+            self.token_sequence_classifier
+                .forward_t(Some(input_tensor.copy()),
+                           None,
+                           None,
+                           None,
+                           None,
+                           false)
+        });
+        let output = output.detach().to(Device::Cpu);
+        let score: Tensor = output.exp() / output.exp().sum1(&[-1], true, Float);
+        let labels_idx = &score.argmax(-1, true);
+
+        let mut tokens: Vec<Token> = vec!();
+        for sentence_idx in 0..labels_idx.size()[0] {
+            let labels = labels_idx.get(sentence_idx);
+            let mut word_idx: u8 = 0;
+            for position_idx in 0..labels.size()[0] {
+                let label_id = labels.int64_value(&[position_idx]);
+                let token = {
+                    let token_id = input_tensor.int64_value(&[sentence_idx, position_idx]);
+                    self.decode_token(token_id, label_id, &score, sentence_idx, position_idx, &mut word_idx)
+                };
+                if let Some(token) = token {
+                    if !ignore_first_label || label_id != 0 {
+                        tokens.push(token);
+                    }
+                }
+            }
+        }
+        tokens
+    }
+
+    fn decode_token(&self, token_id: i64, label_id: i64, score: &Tensor, sentence_idx: i64, position_idx: i64, word_idx: &mut u8) -> Option<Token> {
+        let mut text = match self.tokenizer {
+            TokenizerOption::Bert(ref tokenizer) => Tokenizer::decode(tokenizer, vec!(token_id), false, false),
+            TokenizerOption::Roberta(ref tokenizer) => Tokenizer::decode(tokenizer, vec!(token_id), false, false),
+        };
+        let special_value: bool = match self.tokenizer {
+            //note: we don't count unk as a special value here, as we still want those in the output
+            TokenizerOption::Bert(_) => {
+                text == BertVocab::sep_value() ||
+                text == BertVocab::mask_value() ||
+                text == BertVocab::pad_value() ||
+                text == BertVocab::cls_value()
+            },
+            TokenizerOption::Roberta(_) => {
+                text == RobertaVocab::bos_value() ||
+                text == RobertaVocab::eos_value() ||
+                text == RobertaVocab::sep_value() ||
+                text == RobertaVocab::mask_value() ||
+                text == RobertaVocab::pad_value() ||
+                text == RobertaVocab::cls_value()
+            }
+        };
+        let continuation = !special_value && match self.tokenizer {
```

Comment:
The logic in this section seems to reproduce a lot of tokenization logic. 
- For BERT the continuation tokens should have been removed by the `decode` call earlier?
- I am unsure the code for RoBERTa is correct. The logic behind `text.find(|c:  char| { !c.is_alphabetic() })`

Reply:
Ok, I tested this mainly with RoBERTa, so it could indeed well be that the code for Bert is not right here. The basic idea is that I want to know what tokens are continuation tokens / subtokens.

But all this is probably something that is ideally handled at the tokenizer level as well, rather than after the fact as I'm trying now, but I looked at ``TokenizedInput`` and couldn't seem to deduct which tokens were continuation tokens based on that.

> I am unsure the code for RoBERTa is correct. Could you please explain the logic behind text.find(|c: char| { !c.is_alphabetic() })

It basically excludes anything that is not alphabetic (such as numbers and punctuation) from being classified as a continuation token, and forcing a new word start there.

I guess we should dive into the tokenizer instead and do it there, right?

Reply:
I would suggest to skip the decoding step for now and return simpler tokens that do not contain the `word_index` and `continuation` flag. 

There are a number of improvements that could be made for at the tokenization step, and I would suggest for us to work together on adding the missing features there and come back to the sequence decoding when adequate tools are available for decoding. What do you think?

Reply:
Agreed, done!